tests/IntegrationTests/WebAppUiTests/B2CWebAppCallsWebApiLocally.cs

```diff
@@ -0,0 +1,167 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.Versioning;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Lab.Api;
+using Microsoft.Playwright;
+using TC = Microsoft.Identity.Web.Test.Common.TestConstants;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace WebAppUiTests
+#if !FROM_GITHUB_ACTION
+{
+    // since these tests change environment variables we'd prefer it not run at the same time as other tests
+    [CollectionDefinition(nameof(WebAppCallsApiCallsGraphLocally), DisableParallelization = true)]
+    [Collection("WebAppUiTests")]
+    public class B2CWebAppCallsWebApiLocally : IClassFixture<InstallPlaywrightBrowserFixture>
+    {
+        private const string KeyvaultEmailName = "IdWeb-B2C-user";
+        private const string KeyvaultPasswordName = "IdWeb-B2C-password";
+        private const string KeyvaultClientSecretName = "IdWeb-B2C-Client-ClientSecret";
+        private const string NameOfUser = "unknown";
+        private const uint TodoListClientPort = 5000;
+        private const uint TodoListServicePort = 44332;
+        private const string TraceClassName = "B2CWebAppCallsWebApiLocally";
+        private readonly LocatorAssertionsToBeVisibleOptions _assertVisibleOptions = new() { Timeout = 15000 };
+        private readonly string _devAppPath = Path.Join("DevApps", "B2CWebAppCallsWebApi");
+        private readonly Uri _keyvaultUri = new("https://webappsapistests.vault.azure.net");
+        private readonly ITestOutputHelper _output;
+        private readonly string _testAssemblyPath = typeof(B2CWebAppCallsWebApiLocally).Assembly.Location;
+
+        public B2CWebAppCallsWebApiLocally(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        [Fact]
+        [SupportedOSPlatform("windows")]
+        public async Task ChallengeUser_B2cFlow_LocalApp_ValidEmailPasswordCreds_TodoAppFunctionsCorrectly()
```

Comment:
super nit: B2C is not a flow, so maybe something like `Susi_B2C_LocalAccount_TodoAppFucntionsCorrectly()` is a bit simpler

Reply:
changed